CONTRIBUTING_CODE_COVERAGE.md

```diff
@@ -0,0 +1,110 @@
+
+### Why Playwright?
+
+After exploring our open-source community options such as Jest, Jasmine & Playwright. We decided to move forward with the playwright as it facilitates automated testing, offers cross-browser support, rich enough API with familiar syntax.
+
+## ⚠️ Important Note:
+
+### Before starting your contribution
+
+- create an issue stating which module you will be working on
+- in a given module if tests exist, we do not welcome them as long as it is a logical fix
+- test coverage for example frontend apps are not our top priority
+
+**IMPORTANT: Contributor must create an issue with [Testing Contribution](https://github.com/hiteshchoudhary/apihub/issues/new?assignees=&labels=testing&projects=&template=code_coverage.yaml&title=TESTING%3A+%3Ctitle%3E) issue template.**
+
+This ensures coordination and prevents duplicated efforts.
+
+## Table of Contents
+
+1. [🏁 Getting Started](#getting-started)
+2. [👆🏻 Choosing a Module](#choosing-a-module)
+3. [🗂️ Folder Structure](#folder-structure-main)
+4. [📙 Coding Standards](#coding-standards)
+5. [📝 Dependency Management](#dependency-management)
+6. [📨 Submitting Your Contribution](#submitting-your-contribution)
+
+## Getting Started <a name="getting-started"></a>
+
+### Fork the Repository
+
+Start by forking the FreeAPI project repository to your GitHub account.
+
+### Clone Your Fork
+
+Clone your fork of the repository to your local machine.
+
+```bash
+git clone https://github.com/<your_username>/apihub.git
+cd apihub
+```
+
+### Install Dependencies
+
+Make sure you have the necessary dependencies of FreeAPI installed for the frontend framework or library you plan to use.
+
+Follow this [README.md section](https://github.com/hiteshchoudhary/apihub/blob/main/README.md#-installation) to know more about setting up the FreeAPI environment
+
+## Choosing a Module <a name="choosing-a-module"></a>
+
+Decide which module you want to contribute to. Browse the `/apps`, `/public`, or `/kitchen-sink` modules to explore the available modules and APIs. Read the following section carefully to understand the folder structure you need to follow to increase your chances of getting your PR approved.
+
+## Folder Structure <a name="folder-structure-main"></a>
+
+Follow the specified folder structure for your frontend application (**The folder names must not follow the camel casing to keep things consistent.** _Your actual project code folders may have camel casing_):
+
+```
+ROOT_FOLDER/e2e/{module}/
+```
+
+See the following examples with context for the above structure:
+
+## Example: Todo endpoint testing
+
+Imagine you want to test the todo endpoint that is part of `/apps` module of the FreeAPI project. To keep a consistent folder structure for backtracking name your files with name identifier, but with an extension of `.test.js`.
+
+### Folder Structure:
+
+```
+$ROOT_FOLDER/e2e/{package}/{module}/{file-indicator}.test.js
```

Comment:
We can add folders that are standardise by default with `.gitkeep` included, giving user freedom to only create  `{file-indicator}.test.js` and nothing else. 

You can refer to our frontend [example](https://github.com/hiteshchoudhary/apihub/tree/main/examples) folder for this (I understand that we have some preconfigured test files in `e2e` folder but we can predefined nested folders based on packages and modules) and we can create predefined folders for each package and module with `.gitkeep` file in it to avoid inconsistent packages and modules being injected by contributors.

Would love to hear from you about this approach @jwala-anirudh

Reply:
Only reason behind this was to keep the `e2e` folder structure same as `src` as close as possible in order to understand to which file exactly the test case is being added to.

Reply:
@jwala-anirudh Thanks for the clarification. We will review the folder structure thoroughly to keep our e2e tests consistent before merging.